e2e_test/utils/server.ts

```diff
@@ -0,0 +1,120 @@
+import express from 'express';
+import path from 'path';
+import fs from 'fs/promises';
+import { Server } from 'http';
+
+interface StartServerConfig {
+  basedir: string;
+}
+
+interface StartServerResult {
+  server: Server;
+  port: number;
+}
+
+export const startServer = ({ basedir }: StartServerConfig) => {
+  const app = express();
+  let port: number;
+
+  app.use(function (req, res, next) {
+    let data = '';
+    req.setEncoding('utf8');
+    req.on('data', function (chunk) {
+      data += chunk;
+    });
+
+    req.on('end', function () {
+      req.body = data;
+      next();
+    });
+  });
```

Comment:
what is this for?

Reply:
Removed

Reply:
Reverted. It's needed to parse data in POST routes.